### PR TITLE
feat(QriRef): rename DetailedDatasetRef to VersionInfo, introduce QriRef

### PR DESCRIPTION
**Pull request by b5**
We've been working on the refString -> QriRef -> VersionInfo -> Dataset chain of data enrichment.

This commit is a first-cut to introduce two data models we intend to rely on heavily as we tie more things back to the router

### CODE REVIEW

File: app/actions/mappingFuncs.ts [hunk 1]
```diff
@@ -13,13 +13,13 @@ export function mapRecord (data: any): Record<string, string> {
...
-export function mapDetailedDatasetRef (data: any[]): DetailedDatasetRef[] {
-  return data.map((ref: any) => ref as DetailedDatasetRef)
+export function mapVersionInfo (data: any[]): VersionInfo[] {
+  return data.map((ref: any) => ref as VersionInfo)
 }
 
 // detailedDatasetRefToDataset converts a detailed ref to a sparsely-populated
 // dataset object without using any fetching
-export function detailedDatasetRefToDataset (ddr: DetailedDatasetRef): Dataset {
+export function detailedDatasetRefToDataset (ddr: VersionInfo): Dataset {
```
Comment (ramfox) [inline app/actions/mappingFuncs.ts]: rename to `versionInfoToDataset`

File: app/models/qriRef.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import { RouteComponentProps } from "react-router"
+
+// QriRef models a reference to a user, dataset, or part of a dataset within Qri
+// We use "QriRef" instead of "Ref" to disambiguate with the react "ref"
+// property.
+//
+// QriRef are structured expansions of a reference string, breaking elements of
+// a reference into fields. For example the folling reference string:
```
Comment (ramfox) [inline app/models/qriRef.ts]: "folling" => "following"

File: app/components/network/NetworkHome.tsx [hunk 3]
```diff
@@ -45,21 +49,21 @@ const NetworkHome: React.FC<RouteComponentProps> = ({ history }) => {
...
-          history.push(`/network/${peername}/${name}`)
+          push(`/network/${peername}/${name}`)
         }} data={ds} key={i} />)}
       </div>}
     </div>
   )
 }
 
-export default NetworkHome
+export default withRouter(NetworkHome)
```
Comment (ramfox) [inline app/components/network/NetworkHome.tsx]: just checking that this is the pattern we want. You get `QriRef` s from your container, but `push` functions from `withRouter`?

Comment (b5) [reply app/components/network/NetworkHome.tsx]: still playing with that one, and the answer is I'm not totally sure. I'm working on that now, will follow up with a PR that we can discuss.

**Pull request merged**